pkg/server/types.go

```diff
@@ -16,40 +16,16 @@ limitations under the License.
 
 package server
 
+import (
+	"github.com/heptiolabs/kubernetes-aws-authenticator/pkg/config"
+)
+
 // StaticRoleMapping is a static mapping of a single AWS Role ARN to a
 // Kubernetes username and a list of Kubernetes groups
-type StaticRoleMapping struct {
-	// RoleARN is the AWS Resource Name of the role. (e.g., "arn:aws:iam::000000000000:role/Foo").
-	RoleARN string
-
-	// Username is the Kubernetes username this role will authenticate as (e.g., `mycorp:foo`)
-	Username string
-
-	// Groups is a list of Kubernetes groups this role will authenticate as (e.g., `system:masters`)
-	Groups []string
-}
-
-// Config specifies the configuration for a kubernetes-aws-authenticator server
-type Config struct {
-	// ClusterID is a unique-per-cluster identifier for your
-	// kubernetes-aws-authenticator installation.
-	ClusterID string
-
-	// LocalhostPort is the TCP on which to listen for authentication checks
-	// (on localhost).
-	LocalhostPort int
-
-	// GenerateKubeconfigPath is the output path where a generated webhook
-	// kubeconfig (for `--authentication-token-webhook-config-file`) will be
-	// stored.
-	GenerateKubeconfigPath string
-
-	// StateDir is the directory where generated certificates and private keys
-	// will be stored. You want these persisted between runs so that your API
-	// server webhook configuration doesn't change on restart.
-	StateDir string
+type Server struct {
+	// Config is the whole configuration of kubernetes-aws-authenticator used for valid keys and certs, kubeconfig, and so on
+	config.Config
 
-	// StaticRoleMappings is a list of static mappings from AWS IAM Role to
-	// Kubernetes username+group.
-	StaticRoleMappings []StaticRoleMapping
+	// InitOnServer is set to true when you'd like to generate the key, the certificate, and kubeconfig on server-side
+	InitOnServer bool
```

Comment:
This new flag isn't accessible through the `kubernetes-aws-authenticator server` command.

Do you think it's necessary? I think we could keep the previous behavior of create on-demand, but now you can run `kubernetes-aws-authenticator init` ahead of time if you don't want the on-demand behavior.

Open to keeping it if you have a use case. Maybe we could add a `kubernetes-aws-authenticator server --no-init` flag?